src/main/cpp/blaze_util_freebsd.cc

```diff
@@ -73,16 +73,23 @@ void WarnFilesystemType(const string &output_base) {
 
 string GetSelfPath() {
   char buffer[PATH_MAX] = {};
-  ssize_t bytes = readlink("/proc/curproc/file", buffer, sizeof(buffer));
-  if (bytes == sizeof(buffer)) {
-    // symlink contents truncated
-    bytes = -1;
-    errno = ENAMETOOLONG;
-  }
-  if (bytes == -1) {
-    pdie(blaze_exit_code::INTERNAL_ERROR, "error reading /proc/curproc/file");
+  auto pid = getpid();
+  if (kill(pid, 0) < 0) return "";
+  auto procstat = procstat_open_sysctl();
+  unsigned int n;
+  auto p = procstat_getprocs(procstat, KERN_PROC_PID, pid, &n);
+  if (p) {
+    if (n != 1) {
+      pdie(blaze_exit_code::INTERNAL_ERROR,
+           "expected exactly one process from procstat_getprocs, got %d", n);
+    }
+    auto r = procstat_getpathname(procstat, p, buffer,PATH_MAX);
```

Comment:
I original patch used PATH_MAX - 1，but I changed to PATH_MAX because I see the /usr/bin/procstat do the call like this :https://github.com/freebsd/freebsd/blob/master/usr.bin/procstat/procstat_bin.c#L53，and also here: https://github.com/freebsd/freebsd/blob/master/lib/libproc/proc_create.c#L93. So are you sure this should be MAX_PATH - 1?